torch/_numpy/_funcs_impl.py

```diff
@@ -305,28 +307,39 @@ def linspace(
 
 
 def geomspace(
-    start: ArrayLike,
-    stop: ArrayLike,
+    start: ArrayLikeOrScalar,
+    stop: ArrayLikeOrScalar,
     num=50,
     endpoint=True,
     dtype: Optional[DTypeLike] = None,
     axis=0,
 ):
     if axis != 0 or not endpoint:
         raise NotImplementedError
-    base = torch.pow(stop / start, 1.0 / (num - 1))
-    logbase = torch.log(base)
+
+    all_scalars = all(_dtypes_impl.is_scalar_or_symbolic(x) for x in [start, stop])
+    if all_scalars:
+        exp = math.exp
+        log = math.log
+    else:
+        exp = math.exp
```

Comment:
exp is always `math.exp`, so simply use `math.exp`.

Reply:
Indeed. Now that I actually RTFM, `torch.logspace` only accepts scalars. `numpy.*space` broadcasts, but we do not implement it currently anyway, and this PR is not the place for this enhancement either.

So the last commit adds a comment and always uses `math.exp` and `math.log`